# Chapter 8. Principles of statistical analysis

## Imports 

In [1]:
import DataFrames as pd # pandas in python
import Distributions as dst

## Warning regarding solutions

NO GUARANTEE THAT THE SOLUTIONS WILL WORK OR WORK CORRECTLY! USE THEM AT YOUR OWN RISK!

THE ANSWERS PROVIDED BELOW MAY BE WRONG. USE THEM AT YOUR OWN RISK!

# Exercises

## Exercise 8.2

Eight diabetic patients had plasma glucose levels (mmol/l) measured before and one hour after oral administration of 100 g glucose (Feingold et al., 1989), with the following results

before = [4.67, 4.97, 5.11, 5.17, 5.33, 6.22, 6.50, 7.00]

after = [5.44, 10.11, 8.49, 6.61, 10.67, 5.67, 5.78, 9.89]

(a) Calculate the standard error of the mean change in plasma glucose

(b) On the basis of these data, how many diabetic patients would need to be studied so that the width of the 95% conficence interval for the mean change in plasma glucose level was 0.5 mmol/l? (Assume that the Normal distribution is the appropriate sampling distribution of the change in plasma glucose.)

In [2]:
ex82 = pd.DataFrame((
    ;glucose_before = [4.67, 4.97, 5.11, 5.17, 5.33, 6.22, 6.50, 7.00],
    glucose_after = [5.44, 10.11, 8.49, 6.61, 10.67, 5.67, 5.78, 9.89]
    ))
ex82[:, "change"] = ex82[!, "glucose_after"] - ex82[!, "glucose_before"]
ex82

8×3 DataFrame
 Row │ glucose_before  glucose_after  change  
     │ Float64         Float64        Float64 
─────┼────────────────────────────────────────
   1 │           4.67           5.44     0.77
   2 │           4.97          10.11     5.14
   3 │           5.11           8.49     3.38
   4 │           5.17           6.61     1.44
   5 │           5.33          10.67     5.34
   6 │           6.22           5.67    -0.55
   7 │           6.5            5.78    -0.72
   8 │           7.0            9.89     2.89

In [3]:
function get_mean(xs::Vector{<:Number})::Float64
    return mean::Float64 = sum(xs) / length(xs)
end

function get_sd(xs::Vector{<:Number})::Float64
    mean::Float64 = get_mean(xs)
    return sqrt(sum([(x-mean)^2 for x in xs]) / (length(xs) - 1))
end

get_sd (generic function with 1 method)

In [4]:
function get_sem(xs::Vector{<:Number})::Float64
    return get_sd(xs) / sqrt(length(xs))
end

get_sem (generic function with 1 method)

In [5]:
# here we don't use the formula for two independent samples
# since before and after are dependent, so sem for single sample is OK
get_sem(ex82[!, "change"])

0.8354007539156624

### Ex.8.2(a) Answer

So, SEM of change for glucose is equal to 0.8354...